File: IPL_WEEK_6/D/Main.java
import java.io.*;
import java.util.*;

public class Main {
  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);
    int n = in.nextInt();
    int maxOne = Integer.MIN_VALUE;
    int maxTwo = Integer.MIN_VALUE;
    for(int i = 0; i < n; i++) {
      int one = in.nextInt();
      int two = in.nextInt();
      if(one > two) {
        if(one-two > maxOne) {
          maxOne = one-two;
        }
      }
      else if(two > one) {
        if(two-one > maxTwo) {
          maxTwo = two-one;
        }
      }
    }
    if(maxOne > maxTwo) {
      System.out.println("1 "+maxOne);
    }
    else {
      System.out.println("2 "+maxTwo);
    }
    return;
  }

}

File: IPL_WEEK_6/D/main.cpp
#include<iostream>
using namespace std;

int main() {
	int n;
	int maxOne = 0;
	int maxTwo = 0;
	cin >> n;
	int pOne = 0;
	int pTwo = 0;
	for(int i = 0; i < n; i++) {
		int one, two;
		cin >> one >> two;
		pOne += one;
		pTwo += two;
		if(pOne > pTwo) {
			cout <<pOne-pTwo<<endl;
			if(pOne-pTwo > maxOne) {
				maxOne = pOne-pTwo;
			}
		}
		else if(pTwo > pOne){
			cout<<pTwo-pOne<<endl;
			if(pTwo-pOne > maxTwo) {
				maxTwo = pTwo-pOne;
			}
		}
	}
	if(maxOne > maxTwo) {
		cout << "1 " << maxOne;
	}
	else if(maxTwo > maxOne) {
		cout << "2 " << maxTwo;
	}
	return 0;
}
